ui/js/controllers/bugfiler.js

```diff
@@ -0,0 +1,224 @@
+                $("#modalReftestLog").next().attr("href", reftest);
+            }
+
+            var thisFailure = "";
+            for(var i = 0; i < allFailures.length; i++) {
+                var omittedLeads = ["TEST-UNEXPECTED-FAIL", "PROCESS-CRASH", "TEST-UNEXPECTED-ERROR", "TEST-UNEXPECTED-TIMEOUT"];
+                for(var j=0; j < omittedLeads.length; j++) {
+                    if(allFailures[i][0].search(omittedLeads[j]) >= 0) {
+                        allFailures[i].shift();
+                    }
+                }
+                if(i !== 0) {
+                    thisFailure += "\n";
+                }
+                thisFailure += allFailures[i].join(" | ");
+                $("#modalFailureList");
+            }
+            $("#modalFailureList").val(thisFailure);
+
+            $scope.findProduct();
+        };
+
+        $uibModalInstance.parsedSummary = "";
+        $uibModalInstance.initiate = $scope.initiate;
+        $uibModalInstance.possibleFilename = "";
+
+        /*
+         *  Remove extraneous junk from the start of the summary line
+         *  and try to find the failing test name from what's left
+         */
+        $uibModalInstance.parseSummary = function(summary) {
+            var omittedLeads = ["TEST-UNEXPECTED-FAIL", "PROCESS-CRASH", "TEST-UNEXPECTED-ERROR", "TEST-UNEXPECTED-TIMEOUT"];
+            summary = summary.split(" | ");
+
+            for(var i=0; i < omittedLeads.length; i++) {
+                if(summary[0].search(omittedLeads[i]) >= 0) {
+                    summary.shift();
+                }
+            }
+
+            $uibModalInstance.possibleFilename = summary[0].split("/").pop();
+
+            return [summary, $uibModalInstance.possibleFilename];
+        };
+
+        $scope.toggleFilerSummaryVisibility = function() {
+            $scope.isFilerSummaryVisible = !$scope.isFilerSummaryVisible;
+        };
+
+        $scope.isFilerSummaryVisible = false;
+
+        /*
+         *  Attempt to find a good product/component for this failure
+         */
+        $scope.findProduct = function() {
+            $scope.suggestedProducts = [];
+            var failurePath = $uibModalInstance.parsedSummary[0][0];
+            var failurePathRoot = failurePath.split("/")[0];
+
+            // Look up the product via the root of the failure's file path
+            if(thBugzillaProductObject[failurePathRoot]) {
+                $scope.suggestedProducts.push(thBugzillaProductObject[failurePathRoot][0]);
+            }
+
+            // Look up product suggestions via Bugzilla's api
+            var productSearch = $scope.productSearch;
+
+            if(productSearch) {
+                $http.get("https://bugzilla.mozilla.org/rest/prod_comp_search/" + productSearch + "?limit=5").then(function(request) {
+                    var data = request.data;
+                    $scope.suggestedProducts = [];
+                    for(var i = 0; i < data.products.length;i++) {
```

Comment:
This is totally a style thing for loops, but we often use the lodash library to help here: https://lodash.com/docs#forEach
Or in this case ``_.map`` would be good. It's totally a style thing, so your call.  :)  Just letting you know.  Like you could do:
    $scope.suggestedProducts = _.map(data.products, function(prod) {
        return prod.product + " :: " + prod.component); 
    });